### ISSUE TITLE
Failures on [co19] Roll co19 to 1636313a28df24a9b97154aaa7df3d284a39cb53

### ISSUE THREAD

**Issue by athomas** [dart-lang/co19#1913]
There are new test failures on [[co19] Roll co19 to 1636313a28df24a9b97154aaa7df3d284a39cb53](https://dart-review.googlesource.com/c/sdk/+/287580).

The tests
```
co19/Language/Expressions/Constants/equals_expression_t03 CompileTimeError (expected Pass)
co19/Language/Expressions/Constants/equals_expression_t03 CompileTimeError (expected Pass)
co19/Language/Statements/Switch/syntax_t01 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_element_A02_t01 MissingCompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_element_A01_t01 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_element_A01_t02 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_element_A01_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_element_A01_t03 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_element_A03_t01 MissingCompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_statement_A01_t01 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_statement_A01_t02 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_statement_A01_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_statement_A01_t03 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_in_statement_A03_t01 MissingCompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_for_statement_A02_t01 MissingCompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_if_case_element_A01_t01 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_pattern_if_case_statement_A01_t01 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/execution_switch_expression_A02_t01 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/matching_logical_and_A02_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/matching_relational_A03_t01 RuntimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_arguments_binding_A01_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_arguments_binding_A02_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_arguments_binding_A03_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_class_member_A01_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_class_member_A02_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_class_member_A03_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_fail_arguments_binding_fail_A01_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_fail_arguments_binding_fail_A02_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_fail_arguments_binding_fail_A03_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Subtyping/dynamic/generated/records_fail_arguments_binding_fail_A04_t03 CompileTimeError (expected Pass)
    and 10 more tests
```
are failing on configurations
```
analyzer-asserts-linux
cfe-strong-linux
cfe-weak-linux
dart2js-hostasserts-linux-chrome-unsound
dart2js-hostasserts-linux-d8
dart2js-linux-chrome
dart2wasm-hostasserts-linux-x64-d8
dartk-strong-linux-debug-x64
dartk-strong-linux-release-x64
dartk-weak-asserts-linux-debug-x64
dartk-weak-asserts-linux-release-x64
ddc-linux-chrome
ddc-linux-chrome-unsound
```

**Comment by jakemac53** [comment]
In particular the opt out to `@dart=2.19` is not a viable fix for any test which needs 3.0 features - ie: records/patterns tests. At least some of the failures are related to that, and so instead some classes may be need to marked as mixins instead, and the opt out removed.

If there are any cases where a class is both extended and mixed in, in a test that uses records or patterns, then that test will need to enable the class modifiers experiment and make it a `mixin class`.

**Comment by sgrekhov** [comment]
Subtyping tests failures will be fixed by #1914. All other failures are SDK issues
